### DISCUSSION TITLE
Is it possible to prevent the browser scrollbar disappearing when a select is opened?

### DISCUSSION THREAD

**Discussion by NickWoodward** [adobe/react-spectrum#6612] [Q&A]
When the select components are opened they hide the browser scrollbar. This is fine in firefox, but in chrome it leads to content shifting, which I personally hate. Perhaps this is a mistake on my part, or there's a simple fix, but if not is there any way to prevent this behaviour, or otherwise prevent the content from shifting?

`pl-[calc(100vw-100%)]` (edit: or `w-[100vw]`) on the fixed content stops it from jumping, but the background still covers where the scrollbar was.

See the Header in this demo: https://codesandbox.io/p/devbox/tmsykz?file=%2Fsrc%2FApp.tsx%3A7%2C42

Thanks :)

![image](https://github.com/adobe/react-spectrum/assets/16296682/942ae97a-9da0-457a-a2f3-a9b307976374)

**Comment by snowystinger** [comment]
I think you're running into this https://github.com/adobe/react-spectrum/issues/1216
can you see if any of the workaround mentioned in that issue solve the issue for you?

**Reply by NickWoodward** [reply]
Hi Robert, thanks for the quick reply.

Yeah you're right, I've the same problem as pascalduez. 

I don't know what problems it might cause in other browsers, but a combination of 100vw and overflow hidden on the body has worked for me in chrome and FF

So:
- 100vw on any fixed elements
- 100vw on the html
- overflow-x hidden on the html

Demo: https://codesandbox.io/p/devbox/tmsykz?file=%2Fsrc%2Fstyles.css%3A7%2C22

I'll post it over in that other thread too

Nick

**Reply by NickWoodward** [reply]
Ugh, it was working. Refreshing the page has broken it again :/
I don't understand how that's possible :D

7iomka's reply looks interesting, but I don't really understand it.
I want to say that something like `w-[calc(100vw-(100vw-100%))]` should work, but it doesn't seem to (and is especially unlikely as 100vw now seems buggy)

**Reply by NickWoodward** [reply]
Fingers crossed replacing `100vw` with `pl-[calc(100vw-100%)]` on the fixed element seems to have worked:

[Codesandbox Demo](https://codesandbox.io/p/devbox/tmsykz?file=%2Fsrc%2Fstyles.css%3A8%2C21&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxtg87yn00062a69edbnupjx%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxtg87yn00022a694hmd55i2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxtg87yn00042a690ctox2ke%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxtg87yn00052a691c9fqc3p%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60.518237043906105%252C39.481762956093895%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxtg87yn00022a694hmd55i2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxtg87yn00012a69yrf5ij8d%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxtih51d00022a699d949l78%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A49%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A49%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FItemSelect.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxubupkx00022a69vfg7od89%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A7%252C%2522startColumn%2522%253A52%252C%2522endLineNumber%2522%253A7%252C%2522endColumn%2522%253A52%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxv5r0uv00022a69ndy46ya7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A8%252C%2522startColumn%2522%253A21%252C%2522endLineNumber%2522%253A8%252C%2522endColumn%2522%253A21%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fstyles.css%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxtg87yn00022a694hmd55i2%2522%252C%2522activeTabId%2522%253A%2522clxv5r0uv00022a69ndy46ya7%2522%257D%252C%2522clxtg87yn00052a691c9fqc3p%2522%253A%257B%2522id%2522%253A%2522clxtg87yn00052a691c9fqc3p%2522%252C%2522activeTabId%2522%253A%2522clxtiic2i003l2a69fmokqx1g%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clxtiic2i003l2a69fmokqx1g%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clxtg87yn00042a690ctox2ke%2522%253A%257B%2522id%2522%253A%2522clxtg87yn00042a690ctox2ke%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxtg87yn00032a69gisj4210%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clxtg86fr000hd9f89rtlhhys%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522Development%2522%252C%2522id%2522%253A%2522clxtg8gk500652a69yv83z3oe%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clxtg87yn00032a69gisj4210%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

This time `w-[100vw]` and `overflow-x hidden` on the body
